aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -341,14 +403,33 @@ impl<
                             idx.idx().expect("May not depend on storage version"),
                         ))
                     } else {
-                        Ok(size.value)
+                        Ok(size.value.0)
                     }
                 })
                 .unwrap_or(Err(MVGroupError::Uninitialized)),
             None => Err(MVGroupError::Uninitialized),
         }
     }
 
+    pub fn get_group_size_v2(
+        &self,
+        group_key: &K,
+        txn_idx: TxnIndex,
+        incarnation: Incarnation,
+    ) -> Result<ResourceGroupSize, MVGroupError> {
+        match self.group_sizes.get(group_key) {
+            Some(g) => g
+                .size_entries
+                .range(ShiftedTxnIndex::zero_idx()..ShiftedTxnIndex::new(txn_idx))
```

Comment:
why this has zero_idx while other places don't?

Reply:
no reason, will make it consistent (no need for lower bound) and maybe check invariant that there should never be a lower index/key than Shifted zero idx.

Reply:
Maybe we can have a utility function on group_sizes  to get the upper bound value for the index, so all these range/shifted indices are in one place?